features/cellular/framework/device/CellularDevice.cpp

```diff
@@ -116,9 +117,13 @@ nsapi_error_t CellularDevice::attach_to_network()
 
 nsapi_error_t CellularDevice::create_state_machine()
 {
+    _nw = open_network(_fh);
```

Comment:
This could be inside `if (!_state_machine)` and it should also be closed when `_stat_machine` is deleted.

Reply:
Fixed. _state_machine is deleted in destructor and inheriting class will close all the opened interfaces in it's destructor.